src/commonMain/kotlin/it/krzeminski/snakeyaml/engine/kmp/resolver/CoreScalarResolver.kt

```diff
@@ -43,17 +43,17 @@ class CoreScalarResolver : BaseScalarResolver(
          * Be aware that this regex will also match integers.
          */
         val FLOAT = Regex(
-            "^([-+]?(\\.[0-9]+|[0-9]+(\\.[0-9]*)?)([eE][-+]?[0-9]+)?)" + // float
-                    "|([-+]?\\.(?:inf|Inf|INF))" + // infinity
-                    "|(\\.(?:nan|NaN|NAN))$", // not a number
+            """^(?:([-+]?(\.[0-9]+|[0-9]+(\.[0-9]*)?)([eE][-+]?[0-9]+)?)""" + // float
```

Comment:
Just for reference (I did know about non-capturing groups before - every day brings something new😀):

> unlike capturing groups, it does not memorize the matched text, allowing for better performance and avoiding confusion when the pattern also contains useful capturing groups

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Regular_expressions/Non-capturing_group

Reply:
As a great resource for regex I can also heavily recommend https://www.regular-expressions.info/, especially when it is also about different dialects. :-)